generic_relations/relations.py

```diff
@@ -30,46 +30,27 @@ def __init__(self, serializers, *args, **kwargs):
         """
         super(GenericRelatedField, self).__init__(*args, **kwargs)
         self.serializers = serializers
-        for model, serializer in six.iteritems(self.serializers):
-            # We have to do it, because the serializer can't access a
-            # explicit manager through the GenericForeignKey field on
-            # the model.
-            if hasattr(serializer, 'queryset') and serializer.queryset is None:
-                serializer.queryset = model._default_manager.all()
 
-    def field_to_native(self, obj, field_name):
-        """
-        Delegates to the `to_native` method of the serializer registered
-        under obj.__class__
-        """
-        if not getattr(obj, field_name):
-            # a serializer can't be determined for a nonexistent value
-            return None
-        value = super(GenericRelatedField, self).field_to_native(
-            obj, field_name)
-        serializer = self.determine_deserializer_for_data(value)
-
-        # Necessary because of context, field resolving etc.
-        serializer.initialize(self.parent, field_name)
-        return serializer.to_native(value)
-
-    def to_native(self, value):
-        # Override to prevent the simplifying process of value as present
-        # in `WritableField.to_native`.
-        return value
-
-    def from_native(self, value):
-        # Get the serializer responsible for input resolving
+    def to_internal_value(self, data):
         try:
-            serializer = self.determine_serializer_for_data(value)
+            serializer = self.determine_serializer_for_data(data)
         except ImproperlyConfigured as e:
             raise ValidationError(e)
-        serializer.initialize(self.parent, self.source)
-        return serializer.from_native(value)
+        # Necessary because of context, field resolving etc.
+        serializer.bind(self.field_name, self.parent)
+        return serializer.to_internal_value(data)
+
+    def to_representation(self, instance):
+        serializer = self.determine_deserializer_for_data(instance)
+        # clear source before binding, to avoid an AssertionError
+        serializer.source = None
```

Comment:
Is `to_internal_value` being called before `to_representation`? In that case we're calling `bind` twice which will set `source` in the first call and fail in the `second`.